imgui_widgets.cpp

```diff
@@ -720,6 +720,61 @@ bool ImGui::Button(const char* label, const ImVec2& size_arg)
     return ButtonEx(label, size_arg, ImGuiButtonFlags_None);
 }
 
+bool ImGui::BeginButtonDropDown(const char* label, ImVec2 buttonSize)
+{
+    ImGui::SameLine(0.f, 0.f);
+
+    ImGuiWindow* window = GetCurrentWindow();
+    ImGuiContext& g = *GImGui;
+    const ImGuiStyle& style = g.Style;
+
+    float x = ImGui::GetCursorPosX();
+    float y = ImGui::GetCursorPosY();
+
+    ImVec2 size(20, buttonSize.y);
+    bool pressed = ImGui::Button("##", size);
+
+    // Arrow
+    ImVec2 center(window->Pos.x + x + 10, window->Pos.y + y + buttonSize.y / 2);
+    float r = 8.f;
```

Comment:
I think `r` should scale based on the button size.

Reply:
Maybe a,b and c should also change based on the shape?

Reply:
You're welcome to make that change if you can work out an appropriate value. 

b and c are colour variables. Why should they be influenced by shape?

Reply:
Woops, misread it then.